### PR TITLE
feat(registry): add latexindent formatter

### LINKED ISSUE #261: [New package]: latexindent
### Package name

latexindent

### Package homepage

https://github.com/cmhughes/latexindent.pl

### Languages

latex

### How is this package distributed?

It's distributed in a zip file in the github releases page, the zip comes with `latexindent.pl` and all it's dependencies.

when mason symlinks `latexindent.pl` it should remove the file extension (`latexindent`) as that's how it's normally distributed and how other latex utilities reference it.

Note: this belongs in the `Formatter` category and can be used via `texlab` and `null-ls`

### PR DESCRIPTION
**Pull request by jackwoodside**
Adding [latexindent](https://github.com/cmhughes/latexindent.pl) formatter.

Closes #261

### CODE REVIEW

File: lua/mason-registry/latexindent/init.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    install = function(ctx)
+        github
+            .download_release_file({
+                repo = "cmhughes/latexindent.pl",
+                out_file = platform.is.win and "latexindent.exe" or "latexindent",
+                asset_file = coalesce(
+                    when(platform.is.mac_x64, "latexindent-macos"),
+                    when(platform.is.mac_arm64, "latexindent-macos"),
+                    when(platform.is.linux_arm64, "latexindent-linux"),
+                    when(platform.is.linux_x64, "latexindent-linux"),
```
Comment (williamboman) [inline lua/mason-registry/latexindent/init.lua:26]:
```suggestion
                    when(platform.is.linux_x64_gnu, "latexindent-linux"),
```

File: lua/mason-registry/latexindent/init.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+local github = require "mason-core.managers.github"
+local platform = require "mason-core.platform"
+local std = require "mason-core.managers.std"
+local _ = require "mason-core.functional"
+
+local coalesce, when = _.coalesce, _.when
+
+return Pkg.new {
+    name = "latexindent",
+    desc = [[Perl script to add indentation to LaTeX files.]],
```
Comment (williamboman) [inline lua/mason-registry/latexindent/init.lua:11]:
```suggestion
    desc = _.dedent [[
        latexindent.pl is a perl script to beautify/tidy/format/indent (add horizontal leading space to) code within
        environments, commands, after headings and within special code blocks.
    ]],
```

**Review by williamboman — APPROVED** [review]
Cool! I removed `linux_arm64` because it pointed to a x64 binary. Also added `gnu` requirement to `linux_x64` because the binary is linked to glibc

**Comment by jackwoodside** [comment]
> Cool! I removed `linux_arm64` because it pointed to a x64 binary. Also added `gnu` requirement to `linux_x64` because the binary is linked to glibc

Thanks, I'll keep that in mind for any future additions

**Pull request merged**